### PR TITLE
Use the Zarr's new `getitems()` API

### PR DESCRIPTION
**Pull request by madsbk**
By using the new API in https://github.com/zarr-developers/zarr-python/pull/1131, we do not have to guess whether to read into host or device memory. That is, no more filtering of specify keys like:
```python
 if os.path.basename(fn) in [ 
     zarr.storage.array_meta_key, 
     zarr.storage.group_meta_key, 
     zarr.storage.attrs_key, 
 ]: 
```

Notice, this PR is on hold until Zarr v2.15 is released 

Closes https://github.com/rapidsai/kvikio/issues/119

UPDATE: Zarr v2.15 has been released

### CODE REVIEW

File: python/tests/test_zarr.py
Comment (wence-) [inline python/tests/test_zarr.py:10]: The top-level `cupy` importorskip will mean that if cupy is not available we will never run numpy-based tests. Is that what you want?

File: python/tests/test_zarr.py
Comment (wence-) [inline python/tests/test_zarr.py:32]:
```suggestion
    # Notice, unless using getitems(), GDSStore always returns bytes
```

File: python/tests/test_zarr.py
Comment (wence-) [inline python/tests/test_zarr.py:34]: This equality test doesn't do anything (since there is no assertion).
```suggestion
    assert (xp.frombuffer(b, dtype="u1") == a).all()
```
?

File: python/tests/test_zarr.py
Comment (wence-) [inline python/tests/test_zarr.py:86]: Why are both things not `isinstance(foo, xp.ndarray)`?

File: python/tests/test_zarr.py
Comment (wence-) [inline python/tests/test_zarr.py:105]: Why do you need to slice `z` here? Ah, it's a zarr array so you must slice to get the backing store?

File: python/kvikio/zarr.py
Comment (wence-) [inline python/kvikio/zarr.py:98]: The idiomatic way to do this is...

```
import contextlib

...
with contextlib.ExitStack() as stack:
    for key in keys:
        f = stack.enter_context(kvikio.CuFile(filepath, "r"))
        ret[key] = ..
        io_results.append((f.pread(ret[key]), nbytes)
   for fut, nbytes in io_results:
        nbytes_read = fut.get()
        ...
```

Comment (wence-) [reply python/kvikio/zarr.py:98]: I'll note that both the new code (and my proposed approach) leave dangling futures if the block produces an exception anywhere. I guess that is fine because that propagates and the program is hosed at that point anyway?

**Review by wence- — CHANGES_REQUESTED** [review]
Thanks @madsbk, I think we can do a little bit of cleanup, to avoid repeating ourselves with the minimum supported version.

File: python/tests/test_zarr.py
Comment (madsbk) [reply python/tests/test_zarr.py:89]: They are :)
fixed

File: python/tests/test_zarr.py
Comment (madsbk) [reply python/tests/test_zarr.py:108]: yes

File: python/kvikio/zarr.py
Comment (madsbk) [reply python/kvikio/zarr.py:22]: Fixed :)

File: python/kvikio/zarr.py
Comment (madsbk) [reply python/kvikio/zarr.py:67]: Added docstring

File: python/kvikio/zarr.py
Comment (madsbk) [reply python/kvikio/zarr.py:68]: done

File: python/tests/test_zarr.py
Comment (madsbk) [reply python/tests/test_zarr.py:24]: They should, removed

**Review by wence- — APPROVED** [review]
Thanks Mads.

**Review by wence- — APPROVED** [review]

**Comment by wence-** [comment]
I think we need to wait for the other codeowners to approve too.

**Review by jjacobelli — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks all! 🙏

Great work 🥳

**Pull request merged**
